### ISSUE TITLE
Introduce compatibility with Spring Boot 3

### ISSUE THREAD

**Issue by jk-idealo** [Nylle/JavaFixture#82]
We use annotations like `javax.persistence.Column` to detect primitive constraints (string length, integer range, etc.) when generating random values.

Since Spring Boot 3, the namespaces changed, e.g. `jarkata.persistence.Column`.

If we can't enable support for both, we should consider releasing a new major version which would be incompatible with Boot 2.

**Comment by akutschera** [comment]
If we use the "javax"-dependencies for the javax annotations and the "jakarta"-dependencies for the jakarta annotations, it could work.
Preliminary tests show this also works when someone has
```
<dependency>
  <groupId>jakarta.persistence</groupId>
  <artifactId>jakarta.persistence-api</artifactId>
  <version>2.2.3</version>
</dependency>
<dependency>
  <groupId>jakarta.validation</groupId>
  <artifactId>jakarta.validation-api</artifactId>
  <version>2.0.2</version>
</dependency>
```
even though JavaFixture uses 3.x versions

**Comment by akutschera** [comment]
I can reproduce it now. When I add these to my pom.xml
```
<dependency>
   <groupId>jakarta.persistence</groupId>
   <artifactId>jakarta.persistence-api</artifactId>
   <version>3.1.0</version>
 </dependency>
 <dependency>
   <groupId>jakarta.validation</groupId>
   <artifactId>jakarta.validation-api</artifactId>
   <version>3.0.2</version>
 </dependency>
```
The following test will fail with a NoClassDefFoundError
```
@Test
void testForIssue82() {
    var sut = new Fixture().create( WithEmailAnnotation.class );
    Assertions.assertTrue( sut.email.length() > 0 );
}

public static class WithEmailAnnotation {
    @jakarta.validation.constraints.Email
    String email;
}
```
It seems that the classes are loaded only when the code actually enters the line where we need it.
I have no idea how to reproduce that in a unit test.

**Comment by jk-idealo** [comment]
Fixed in version [2.9.4](https://github.com/Nylle/JavaFixture/releases/tag/2.9.4).
